compute_sdk/globus_compute_sdk/sdk/mpi_function.py

```diff
@@ -0,0 +1,76 @@
+import typing as t
+
+from .shell_function import ShellFunction, ShellResult
+
+
+class MPIFunction(ShellFunction):
+    """MPIFunction extends ShellFunction, as a thin wrapper that adds an
+    MPI launcher prefix to the BashFunction command.
+    """
+
+    def __init__(
+        self,
+        cmd: str,
+        stdout: t.Optional[str] = None,
+        stderr: t.Optional[str] = None,
+        walltime: t.Optional[float] = None,
+        snippet_lines=1000,
+    ):
+        """Initialize a ShellFunction
+
+        Parameters
+        ----------
+        cmd: str
+             formattable command line to execute. For e.g:
+             "lammps -in {input_file}" where {input_file} is formatted
+             with kwargs passed at call time.
+
+        stdout: str | None
+            file path to which stdout should be captured
+
+        stderr: str | None
+            file path to which stderr should be captures
+
+        walltime: float | None
+            duration in seconds after which the command should be interrupted
+
+        snippet_lines: int
+            Number of lines of stdout/err to capture,
+            default=1000
+
+        """
+        self.cmd = cmd
+        self.stdout = stdout
+        self.stderr = stderr
+        self.walltime = walltime
+        self.snippet_lines = snippet_lines
```

Comment:
Would it be wise to verify that these values are at least sane?  For example are negative values meaningful?

Reply:
Probably overkill. I'm feeding walltime straight to `proc.wait(<timeout>)`, and it treats values less that 0 as 0.

Reply:
Fixed.